tests/fuzz/analyzer_fuzzer.go

```diff
@@ -0,0 +1,174 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint: golint
+package fuzz
+
+import (
+	"io/ioutil"
+	"os"
+	"time"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+	"istio.io/pkg/log"
+
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/analyzers"
+	"istio.io/istio/galley/pkg/config/analysis/local"
+	"istio.io/istio/galley/pkg/config/scope"
+	"istio.io/istio/pkg/config/schema"
+	"istio.io/istio/pkg/config/schema/collection"
+)
+
+var (
+	availableAnalyzers = analyzers.All()
+)
+
+// createRandomConfigFile creates a single fuzzed config file
+func createRandomConfigFile(f *fuzz.ConsumeFuzzer) (string, error) {
+	data, err := f.GetBytes()
+	if err != nil {
+		return "nobytes", err
+	}
+	tmpfile, err := ioutil.TempFile("", "example")
+	if err != nil {
+		return "nofile", err
+	}
+	if _, err := tmpfile.Write(data); err != nil {
+		return "nofile", err
+	}
+	if err := tmpfile.Close(); err != nil {
+		return "nofile", err
+	}
+	return tmpfile.Name(), nil
+}
+
+// createRandomConfigFiles creates a slice of fuzzed config files
+func createRandomConfigFiles(f *fuzz.ConsumeFuzzer) ([]local.ReaderSource, error) {
+	var files []local.ReaderSource
+
+	numberOfFiles, err := f.GetInt()
+	if err != nil {
+		return files, err
+	}
+	maxFiles := numberOfFiles % 10
+
+	for i := 0; i < maxFiles; i++ {
+		configFile, err := createRandomConfigFile(f)
+		if err != nil {
+			return files, err
+		}
+		defer os.Remove(configFile)
+		of, err := os.Open(configFile)
+		if err != nil {
+			return files, err
+		}
+		files = append(files, local.ReaderSource{Name: configFile, Reader: of})
```

Comment:
Should we use strings.NewReader instead of a real file?

Reply:
Yes, that should have a positive impact on performance. Updated fuzzer with `bytes.NewReader`